cibyl/sources/jenkins.py

```diff
@@ -60,64 +61,85 @@ def __init__(self, url: str, username: str = None, token: str = None,
         self.client._session.verify = cert
 
     @safe_request
-    def get_jobs(self, get_builds: bool = False, **kwargs):
+    def get_jobs(self, **kwargs):
         """
             Get all jobs from jenkins server.
 
-            :param get_builds: Whether to get info about the jobs' builds
-            :type get_builds: bool
-
-            :returns: All jobs from jenkins server, as dictionaries of _class,
-            name, fullname, url, color
+            :returns: List of Job objects from jenkins server
             :rtype: list
         """
-        if get_builds:
-            return self.client.get_info(query=self.jobs_builds_query)["jobs"]
         jobs_arg = kwargs.get('jobs')
-        if jobs_arg:
-            jobs_found = []
-            for job in jobs_arg.value:
-                LOG.debug("querying %s for job %s", self.name, job)
-                jobs_found.extend(self.client.get_job_info_regex(pattern=job))
-            return jobs_found
-
-        return self.client.get_info(query=self.jobs_query)["jobs"]
-
-    # pylint: disable=no-self-use
-    def populate_jobs(self, system, jobs: list[dict]):
+        if not jobs_arg:
+            jobs_arg = [""]
+        jobs_result = {}
+
+        # default case, where user wants all jobs
+        jobs_found = []
+        for job in jobs_arg:
+            jobs_found = self.client.run_script(
```

Comment:
I'm getting 403 from Jenkins while using the script. I think you need elevated permissions to access the entrypoint for running scripts?

Reply:
Oh no. Can you paste the full error?